knack/config.py

```diff
@@ -47,36 +47,126 @@ def __init__(self, config_dir=None, config_env_var_prefix=None, config_file_name
+                last_ex = ex
+
+        if fallback is _UNSET:
+            raise last_ex  # pylint:disable=raising-bad-type
+        return fallback
+
+    def items(self, section, use_local_config=False):
+        import re
+        pattern = self.env_var_name(section, '.+')
+        candidates = [(k.split('_')[-1], os.environ[k], k) for k in os.environ.keys() if re.match(pattern, k)]
+        result = {c[0]: c for c in candidates}
+        for config in self.config_file_chain if use_local_config else self.config_file_chain[-1:]:
+            try:
+                entries = config.items(section)
+                for name, value in entries:
+                    if name not in result:
+                        result[name] = (name, value, config.config_path)
+            except (configparser.NoSectionError, configparser.NoOptionError):
+                pass
+        return [{'name': name, 'value': value, 'source': source} for name, value, source in result.values()]
+
+    def getint(self, section, option, fallback=_UNSET, use_local_config=False):
+        return int(self.get(section, option, fallback, use_local_config))
+
+    def getfloat(self, section, option, fallback=_UNSET, use_local_config=False):
+        return float(self.get(section, option, fallback, use_local_config))
+
+    def getboolean(self, section, option, fallback=_UNSET, use_local_config=False):
+        val = str(self.get(section, option, fallback, use_local_config))
+        if val.lower() not in CLIConfig._BOOLEAN_STATES:
+            raise ValueError('Not a boolean: {}'.format(val))
+        return CLIConfig._BOOLEAN_STATES[val.lower()]
 
-    def get(self, section, option, fallback=_UNSET):
+    def set_value(self, section, option, value, use_local_config=False):
+        if use_local_config:
+            current_config_dir = os.path.join(os.getcwd(), os.path.basename(self.config_dir))
+            config_file_path = os.path.join(current_config_dir, os.path.basename(self.config_path))
+            if config_file_path == self.config_file_chain[0].config_path:
+                self.config_file_chain[0].set_value(section, option, value)
+            else:
+                config = _ConfigFile(current_config_dir, config_file_path)
+                config.set_value(section, option, value)
+                self.config_file_chain.insert(0, config)
+        else:
+            self.config_file_chain[-1].set_value(section, option, value)
         try:
-            env = self.env_var_name(section, option)
-            return os.environ[env] if env in os.environ else self.config_parser.get(section, option)
-        except (configparser.NoSectionError, configparser.NoOptionError):
-            if fallback is _UNSET:
-                raise
-            return fallback
+            self.config_parser.add_section(section)
+        except configparser.DuplicateSectionError:
+            pass
+        self.config_parser.set(section, option, value)
```

Comment:
do we still need this? I thought `_ConfigFile.set_value` should handle it, no?

Reply:
Reomved